notebooks/config.ipynb

```diff
@@ -0,0 +1,1046 @@
+      "File \u001b[0;32m~/NOAA/uwtools/src/uwtools/config/formats/ini.py:21\u001b[0m, in \u001b[0;36mINIConfig.__init__\u001b[0;34m(self, config)\u001b[0m\n\u001b[1;32m     17\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__init__\u001b[39m(\u001b[38;5;28mself\u001b[39m, config: Union[\u001b[38;5;28mdict\u001b[39m, Optional[Path]] \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[1;32m     18\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m     19\u001b[0m \u001b[38;5;124;03m    :param config: Config file to load (None => read from stdin), or initial dict.\u001b[39;00m\n\u001b[1;32m     20\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m---> 21\u001b[0m     \u001b[38;5;28;43msuper\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[38;5;21;43m__init__\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mconfig\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     22\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_parse_include()\n",
+      "File \u001b[0;32m~/NOAA/uwtools/src/uwtools/config/formats/base.py:36\u001b[0m, in \u001b[0;36mConfig.__init__\u001b[0;34m(self, config)\u001b[0m\n\u001b[1;32m     34\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdata \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_load(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_config_file)\n\u001b[1;32m     35\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_get_depth_threshold() \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_depth \u001b[38;5;241m!=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_get_depth_threshold():\n\u001b[0;32m---> 36\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m UWConfigError(\n\u001b[1;32m     37\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCannot instantiate depth-\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m with depth-\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m config\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     38\u001b[0m         \u001b[38;5;241m%\u001b[39m (\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_get_depth_threshold(), \u001b[38;5;28mtype\u001b[39m(\u001b[38;5;28mself\u001b[39m)\u001b[38;5;241m.\u001b[39m\u001b[38;5;18m__name__\u001b[39m, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_depth)\n\u001b[1;32m     39\u001b[0m     )\n",
+      "\u001b[0;31mUWConfigError\u001b[0m: Cannot instantiate depth-2 INIConfig with depth-1 config"
+     ]
+    }
+   ],
+   "source": [
+    "config.get_ini_config(\n",
+    "    config={\"greeting\":\"Salutations\", \"recipient\":\"Mars\"}\n",
+    ")"
+   ]
```

Comment:
![image](https://github.com/user-attachments/assets/aca0c1b9-5987-4463-a25c-7761d3f18ae7)

We could alternatively demonstrate errors like this, which looks a lot less messy and I think draws the user's attention to the important part. This is actually more like what an API user would do, too: They'd want to catch exceptions thrown in `uwtools` and handle them, rather than letting them crash their program. What do you think?

This also makes me think that we probably need to expose `uwtools` exceptions via the API, too, so that users can `import` them and specifically catch them. Here I'm catching a generic `Exception` object, but a linter would complain about this, and a user probably wouldn't really want to cast such a wide net, and catch other exceptions like `KeyError` or `DivideByZeroError`. I think it's still ok for a demonstration, but ideally we could `from uwtools.api.exceptions import UWConfigError` and then `except UWConfigError as e:`.

Reply:
I like this style of demonstrating errors a lot! Besides cutting down on the length of the notebook and drawing attention to the important part, I think it will also make unit testing easier too. I'll update the error demo cells to look like this.